src/background.js

```diff
@@ -0,0 +1,15 @@
+//TODO Change sample code to acceot scripting and option to apply on any page
+chrome.runtime.onInstalled.addListener(function() {
+    chrome.storage.sync.set({color: '#3aa757'}, function() {
+      console.log("The color is green.");
+    });
+    chrome.declarativeContent.onPageChanged.removeRules(undefined, function() {
+        chrome.declarativeContent.onPageChanged.addRules([{
+          conditions: [new chrome.declarativeContent.PageStateMatcher({
+            pageUrl: {hostEquals: 'developer.chrome.com'},
+          })
+          ],
+              actions: [new chrome.declarativeContent.ShowPageAction()]
+        }]);
+      });
+  });
```

Comment:
So your above code (If I'm reading it correctly) will store a colour, and then do something with the icon to the right of the address bar.

This is a snippet I used in my background.js file:

```
// listens for the address bar icon to be clicked
chrome.browserAction.onClicked.addListener(function (tab) {
  // for the current tab, inject the "google-alert.js" file & execute it
  chrome.tabs.executeScript(tab.ib, { file: "./scripts/google-alert.js" });
});
```

You would need to figure out how to pass the code instead of using files, but I think you might be close to getting this working